server/sonar-process-monitor/src/main/java/org/sonar/application/process/AbstractProcessMonitor.java

```diff
@@ -24,80 +24,54 @@
 import java.io.InputStream;
 import java.util.concurrent.TimeUnit;
 import javax.annotation.Nullable;
-import org.sonar.process.ProcessCommands;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import static java.util.Objects.requireNonNull;
+abstract class AbstractProcessMonitor implements ProcessMonitor {
 
-class ProcessMonitorImpl implements ProcessMonitor {
+  private static final Logger LOG = LoggerFactory.getLogger(AbstractProcessMonitor.class);
 
-  private final Process process;
-  private final ProcessCommands commands;
+  protected final Process process;
 
-  ProcessMonitorImpl(Process process, ProcessCommands commands) {
-    this.process = requireNonNull(process, "process can't be null");
-    this.commands = requireNonNull(commands, "commands can't be null");
+  protected AbstractProcessMonitor(Process process) {
+    this.process = process;
   }
 
-  @Override
   public InputStream getInputStream() {
     return process.getInputStream();
   }
 
-  @Override
   public void closeStreams() {
     closeQuietly(process.getInputStream());
     closeQuietly(process.getOutputStream());
     closeQuietly(process.getErrorStream());
   }
 
-  @Override
+  private static void closeQuietly(@Nullable Closeable closeable) {
+    try {
+      if (closeable != null) {
+        closeable.close();
+      }
+    } catch (IOException ignored) {
+      // ignore
+    }
+  }
+
   public boolean isAlive() {
     return process.isAlive();
   }
 
-  @Override
   public void destroyForcibly() {
     process.destroyForcibly();
   }
 
-  @Override
   public void waitFor() throws InterruptedException {
+    LOG.warn("Daniels Debugging: waitFor ... "+this);
     process.waitFor();
+    LOG.warn("Daniels Debugging: waitFor done "+this);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2629)